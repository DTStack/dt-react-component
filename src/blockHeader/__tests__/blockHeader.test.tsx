import React from 'react';
import { cleanup, fireEvent, render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import BlockHeader, { type SizeType } from '../index';

const props = {
    title: '标题1',
};
const props2 = {
    title: '标题2',
    beforeTitle: <span>Icon</span>,
    afterTitle: '说明文字',
    addonAfter: <div className="test-button-after">新增按钮</div>,
    size: 'small' as SizeType,
    titleRowClassName: 'test-row-className',
    titleClassName: 'test-title-className',
};
const props3 = {
    title: 'hover',
    tooltip: 'hover 展示',
};
const props4 = {
    title: 'hover',
    tooltip: 'hover 展示',
    afterTitle: '我的优先级更高',
};
const prefixCls = 'dtc-block-header';

describe('test BlockHeader render', () => {
    afterEach(() => {
        cleanup();
    });
    test('should render BlockHeader success render', () => {
        const wrapper = render(<BlockHeader {...props} />);
        expect(wrapper).toMatchSnapshot();
    });
    test('should render BlockHeader default BlockHeader render', () => {
        const { getByText, rerender } = render(<BlockHeader {...props} />);
        expect(getByText(props.title)).toBeTruthy();
        rerender(<BlockHeader title="标题2" />);
        expect(getByText('标题2')).toBeTruthy();
    });
    test('should render BlockHeader props default in BlockHeader', () => {
        const { container } = render(<BlockHeader title="测试" showBackground />);
        const wrap = container.firstChild;
        expect(wrap!.firstChild!.firstChild!.firstChild).toHaveClass(`${prefixCls}-before-title`);
        fireEvent.click(document.getElementsByClassName(`${prefixCls}-title-row`)[0]);
    });
    test('should render BlockHeader with different props', () => {
        const { container, getByText } = render(<BlockHeader {...props2} />);
        const wrap = container.firstChild;
        expect(wrap).toHaveClass(`${prefixCls}`);
        expect(wrap!.lastChild).toHaveClass(`${prefixCls}-content`);
        expect(wrap!.firstChild).toHaveClass(`test-row-className`);
        expect(getByText('标题2')).toHaveClass('test-title-className');
        expect(getByText('说明文字')).toHaveClass(`${prefixCls}-after-title`);
        expect(getByText('新增按钮')).toHaveClass(`test-button-after`);
        expect(getByText('Icon')).toBeTruthy();
    });
    test('should render BlockHeader test click event', () => {
        const onChange = jest.fn();
        const { getByText } = render(
            <BlockHeader onChange={onChange} title="测试">
                <div>1111</div>
            </BlockHeader>
        );
        expect(getByText('收起')).toBeTruthy();
        fireEvent.click(document.getElementsByClassName(`${prefixCls}-title-row`)[0]);
        expect(getByText('展开')).toBeTruthy();
        expect(onChange).toHaveBeenCalledTimes(1);
    });
    test('should render expanded and collapsed BlockHeader normally if the onChange event is not set', () => {
        const { getByText } = render(
            <BlockHeader title="测试">
                <div>Hello World!</div>
            </BlockHeader>
        );
        expect(getByText('收起')).toBeTruthy();
        fireEvent.click(document.getElementsByClassName(`${prefixCls}-title-row`)[0]);
        expect(getByText('展开')).toBeTruthy();
    });
    test('should render BlockHeader with different props', () => {
        const { container, getByText } = render(<BlockHeader {...props2} />);
        const wrap = container.firstChild;
        expect(wrap).toHaveClass(`${prefixCls}`);
        expect(wrap!.lastChild).toHaveClass(`${prefixCls}-content`);
        expect(wrap!.firstChild).toHaveClass(`test-row-className`);
        expect(getByText('标题2')).toHaveClass('test-title-className');
        expect(getByText('说明文字')).toHaveClass(`${prefixCls}-after-title`);
        expect(getByText('Icon')).toBeTruthy();
    });
    test('should render BlockHeader className when isSmall is small', () => {
        const props = { title: '测试1', showBackground: false };
        const { container } = render(<BlockHeader {...props} />);
        const wrap = container.firstChild;
        expect(wrap!.firstChild).not.toHaveClass(`background`);
    });
    test('should render BlockHeader className when isSmall is small', () => {
        const { container, getByText } = render(<BlockHeader {...props2} />);
        const wrap = container.firstChild!;
        expect(wrap).toHaveClass(`${prefixCls}`);
        expect(wrap.lastChild).toHaveClass(`${prefixCls}-content`);
        expect(wrap.firstChild).toHaveClass(`test-row-className`);
        expect(getByText('标题2')).toHaveClass('test-title-className');
        expect(getByText('说明文字')).toHaveClass(`${prefixCls}-after-title`);
        expect(getByText('Icon')).toBeTruthy();
    });

    test('should render BlockHeader tooltip success', () => {
        const { container } = render(<BlockHeader {...props3} />);
        const wrap = container.firstChild!;
        const afterTitleWrap = wrap.firstChild!.firstChild!.lastChild;
        expect(afterTitleWrap!.firstChild).toHaveClass('anticon-question-circle');
    });

    test('should render BlockHeader tooltip and afterTitle success', () => {
        const { container } = render(<BlockHeader {...props4} />);
        const wrap = container.firstChild!;
        const afterTitleWrap = wrap.firstChild!.firstChild!.lastChild;
        expect(afterTitleWrap).toHaveTextContent('我的优先级更高');
    });
    test('should render BlockHeader correct dom length', () => {
        const { container } = render(<BlockHeader title="分类级别" beforeTitle="" />);
        const titleBoxWrap = container.firstChild!.firstChild!.firstChild;
        expect(titleBoxWrap!.childNodes.length).toEqual(1);

        const { container: container1 } = render(
            <BlockHeader title="分类级别" afterTitle="测试" />
        );
        const titleBoxWrap1 = container1.firstChild!.firstChild!.firstChild;
        expect(titleBoxWrap1!.childNodes.length).toEqual(3);
    });
    test('should render BlockHeader correct margin-bottom', () => {
        const { container: noStyle } = render(<BlockHeader title="分类级别" beforeTitle="" />);
        expect(noStyle.querySelector('.dtc-block-header')).not.toHaveAttribute('style');
        const { container: defaultBottom } = render(
            <BlockHeader title="分类级别" beforeTitle="" />
        );
        expect(defaultBottom.querySelector('.dtc-block-header')).toHaveStyle({ marginBottom: 16 });
        const { container: customizeBottom } = render(
            <BlockHeader title="分类级别" beforeTitle="" hasBottom spaceBottom={10} />
        );
        expect(customizeBottom.querySelector('.dtc-block-header')).toHaveStyle({
            marginBottom: 10,
        });
    });
});
