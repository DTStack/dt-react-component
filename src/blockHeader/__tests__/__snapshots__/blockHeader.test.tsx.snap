// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test BlockHeader render should render BlockHeader success render 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="dtc-block-header"
      >
        <div
          class="dtc-block-header__title dtc-block-header__title--middle dtc-block-header__title--background"
        >
          <div
            class="title__box"
          >
            <div
              class="title__addon-before"
            >
              <div
                class="title__addon-before--middle"
              />
            </div>
            <div
              class="title__text"
            >
              标题1
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="dtc-block-header"
    >
      <div
        class="dtc-block-header__title dtc-block-header__title--middle dtc-block-header__title--background"
      >
        <div
          class="title__box"
        >
          <div
            class="title__addon-before"
          >
            <div
              class="title__addon-before--middle"
            />
          </div>
          <div
            class="title__text"
          >
            标题1
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
